client/rpcflags/rpcflags_test.go

```diff
@@ -0,0 +1,134 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rpcflags
+
+import (
+	"context"
+	"encoding/pem"
+	"flag"
+	"strings"
+	"testing"
+
+	"github.com/google/trillian"
+	"github.com/google/trillian/testonly/integration"
+	"github.com/google/trillian/testonly/setup"
+	"github.com/google/trillian/util/flagsaver"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+func TestNewClientDialOptionsFromFlagsWithTLSCertFileNotSet(t *testing.T) {
+	// Set up Trillian servers
+	const numSequencers = 2
+	serverOpts := []grpc.ServerOption{}
+	clientOpts := []grpc.DialOption{grpc.WithInsecure()}
+	logEnv, err := integration.NewLogEnvWithGRPCOptions(context.Background(), numSequencers, serverOpts, clientOpts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer logEnv.Close()
+
+	dialOpts, err := NewClientDialOptionsFromFlags()
+	if err != nil {
+		t.Errorf("Got an unexpected error: %v", err)
+	}
+
+	// Check that the returned dial options can be used to connect and make
+	// requests against the Trillian services created above.
+	conn, err := grpc.Dial(logEnv.Address, dialOpts...)
+	if err != nil {
+		t.Errorf("failed to dial %v: %v", logEnv.Address, err)
+	}
+	defer conn.Close()
+
+	adminClient := trillian.NewTrillianAdminClient(conn)
+	_, err = adminClient.ListTrees(context.Background(), &trillian.ListTreesRequest{})
+	if err != nil {
+		t.Errorf("failed to request trees from the Admin Server: %v", err)
+	}
+}
+
+func TestNewClientDialOptionsFromFlagsWithTLSCertFileMissing(t *testing.T) {
+	defer flagsaver.Save().Restore()
+	err := flag.Set("tls_cert_file", "/a/missing/file")
+	if err != nil {
+		t.Errorf("Failed to set flag: %v", err)
+	}
+
+	dialOpts, err := NewClientDialOptionsFromFlags()
+	if err == nil || !strings.Contains(err.Error(), "no such file or directory") {
+		t.Errorf("Expected to get an error due to the file not being found, instead got: %v", err)
+	}
+
+	if dialOpts != nil {
+		t.Errorf("Expected returned dialOpts to be nil, instead got: %v", dialOpts)
+	}
+}
+
+func TestNewClientDialOptionsFromFlagsWithTLSCertFileSet(t *testing.T) {
+	// Create new TLS certificates for the test services, and write the Client
+	// certificate to a file (so we can refer to it using the flag).
+	crtFile, cleanupCrtFile := setup.TempFile(t, "test.crt.")
+	defer cleanupCrtFile()
+
+	tlsCert := setup.NewTLSCertificate(t)
+
+	// Certificate file and client dial options.
+	err := pem.Encode(crtFile, &pem.Block{Type: "CERTIFICATE", Bytes: tlsCert.Certificate[0]})
+	if err != nil {
+		t.Fatalf("Failed to encode the test TLS certificate %v", err)
+	}
+	clientCreds, err := credentials.NewClientTLSFromFile(crtFile.Name(), "")
+	if err != nil {
+		t.Fatalf("Failed to get credentials: %v", err)
+	}
+
+	// Set up Trillian servers (with TLS enabled)
+	const numSequencers = 2
```

Comment:
I guess you don't actually need sequencers, could you set this to zero to speed the test up/use fewer resources?

Reply:
Done.